codec-dns/src/main/java/io/netty/handler/codec/dns/DohResponseDecoder.java

```diff
@@ -0,0 +1,38 @@
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.FullHttpResponse;
+
+import java.net.SocketAddress;
+
+public class DohResponseDecoder extends SimpleChannelInboundHandler<FullHttpResponse> {
+
+    private final DnsResponseDecoder<SocketAddress> responseDecoder;
+
+    public DohResponseDecoder() {
+        responseDecoder = new DnsResponseDecoder<SocketAddress>(DnsRecordDecoder.DEFAULT) {
+            @Override
+            protected DnsResponse newResponse(SocketAddress sender, SocketAddress recipient, int id, DnsOpCode opCode,
+                                              DnsResponseCode responseCode) {
+                return new DefaultDnsResponse(id, opCode, responseCode);
+            }
+        };
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpResponse msg) throws Exception {
```

Comment:
I guess you need to at least handle http error codes here?

Reply:
I have handled the exceptions. Please take a look and give me more advice on handling additional exceptions.

Reply:
Hi @mostroverkhov , I've modified the PR and solved its problems. Could you please review that again? Thanks